vpr/src/route/rr_graph_storage.cpp

```diff
@@ -709,3 +709,68 @@ t_rr_graph_view t_rr_graph_storage::view() const {
         vtr::make_const_array_view_id(edge_dest_node_),
         vtr::make_const_array_view_id(edge_switch_));
 }
+
+void t_rr_graph_storage::reorder(const vtr::vector<RRNodeId, RRNodeId>& order,
+                                 const vtr::vector<RRNodeId, RRNodeId>& inverse_order) {
+    {
+        auto old_node_storage = node_storage_;
+
+        // Reorder nodes
+        for (size_t i = 0; i < node_storage_.size(); i++) {
+            auto n = RRNodeId(i);
+            VTR_ASSERT(n == inverse_order[order[n]]);
+            node_storage_[order[n]] = old_node_storage[n];
+        }
+    }
+    {
+        auto old_node_first_edge = node_first_edge_;
+        auto old_edge_src_node = edge_src_node_;
+        auto old_edge_dest_node = edge_dest_node_;
+        auto old_edge_switch = edge_switch_;
+        RREdgeId cur_edge(0);
+
+        // Reorder edges by source node
+        for (size_t i = 0; i < node_storage_.size(); i++) {
```

Comment:
Why is this loop required?  Why not simply map src and dest nodes from old -> new and re-sort?

Reply:
Why use an O(N log N) algorithm (sort) instead of an O(N) one?

Reply:
Also, this would not preserve partitioning.